src/uct/ib/base/ib_iface.c

```diff
@@ -820,12 +820,17 @@ static ucs_status_t uct_ib_iface_init_pkey(uct_ib_iface_t *iface,
 
         if ((config->pkey == UCS_HEXUNITS_AUTO) ||
             /* take only the lower 15 bits for the comparison */
-            ((pkey & UCT_IB_PKEY_PARTITION_MASK) == config->pkey)) {
-            if (!(pkey & UCT_IB_PKEY_MEMBERSHIP_MASK) &&
-                /* limited PKEY has not yet been found */
-                (lim_pkey == UCT_IB_ADDRESS_INVALID_PKEY)) {
-                lim_pkey_index = pkey_index;
-                lim_pkey       = pkey;
+            ((pkey & UCT_IB_PKEY_PARTITION_MASK) == config->pkey) ||
+            /* RoCE uses PKEY index 0 automatically */
+            uct_ib_iface_is_roce(iface)) {
+            if (/* do not check membership mask for RoCE */
```

Comment:
i'd also avoid such comment, because it just repeats what he code does.

Reply:
removed